CMakeLists.txt

```diff
@@ -23,7 +23,7 @@ find_package(Threads REQUIRED)
 
 if (GTN_BUILD_CUDA)
   include_directories(${CUDA_INCLUDE_DIRS})
-  add_definitions(-DCUDA)
+  add_definitions(-D_CUDA_)
```

Comment:
is this intentional ?

Reply:
Yes, because I am using the name `Device::CUDA` for the choosing the CUDA device and this was confusing the preprocessor.

Reply:
Should it be prefixed with the project name, e.g., `GTN_CUDA`?